pkg/cloud/aws/commands/run.go

```diff
@@ -124,13 +125,58 @@ func Run(ctx context.Context, opt flag.Options) error {
 			return fmt.Errorf("aws scan error: %w", err)
 		}
 	}
-	r := report.New(cloud.ProviderAWS, opt.Account, opt.Region, results.GetFailed(), opt.Services)
 
 	log.Logger.Debug("Writing report to output...")
+	if len(opt.Compliance) > 0 && opt.ReportFormat == "summary" {
```

Comment:
@simar7 do we want to produce report == all ? 
should we change the condition to `if len(opt.Compliance) > 0 && (opt.ReportFormat == "summary" || opt.ReportFormat == "all" {` ?

Reply:
Yes I agree, that makes sense. Updated here https://github.com/aquasecurity/trivy/pull/3130/commits/feb12ccc0f921188150279759d034802990c29d4